src/test/groovy/com/synacy/gradprogram/spock/exercise/DeliveryServicePec.groovy

```diff
@@ -0,0 +1,49 @@
+package com.synacy.gradprogram.spock.exercise
+
+import spock.lang.Specification
+
+
+class DeliveryServiceSpec extends Specification {
+
+    DeliveryService deliveryService
+    DeliveryRequestRepository deliveryRequestRepository = Mock(DeliveryRequestRepository)
+    DateUtils dateUtils = Mock(DateUtils)
+
+    void setup(){
+        deliveryService = new DeliveryService(deliveryRequestRepository, dateUtils)
+    }
+
+    def "Should Create Delivery Request"() {
+        given:
+        Order order = new Order()
+        dateUtils.getCurrentDate() >> new Date()
+
+        when:
+        DeliveryRequest result = deliveryService.createDelivery(order)
+        then:
+        1 * dateUtils.getCurrentDate() >> new Date()
+        1 * deliveryRequestRepository.saveDeliveryRequest(_)
+
+        and:
+        result.orderId == order.id
+        result.courier != null
+        result.deliveryDate != null
```

Comment:
We should test with actual values here and not just null checks. Additionally, we should check the different scenarios that is tied to the order's total cost.